src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -16,7 +16,7 @@ internal class IPAddressParser
 
         internal static unsafe IPAddress Parse(ReadOnlySpan<char> ipSpan, bool tryParse)
         {
-            if (ipSpan.IndexOf(':') >= 0)
+            if (ipSpan.Contains(':'))
```

Comment:
There are other areas we should fix up:
https://github.com/dotnet/corefx/blob/60a6bbdd646b214ea71b5bfbb0a55eb41eb5eae2/src/Common/src/Interop/Unix/System.Native/Interop.ReadDir.cs#L53
https://github.com/dotnet/corefx/blob/60a6bbdd646b214ea71b5bfbb0a55eb41eb5eae2/src/Common/src/System/Memory/FixedBufferExtensions.cs#L27
...

Do you plan to fixup the string.IndexOf calls here as well?

For example:
https://github.com/dotnet/corefx/blob/60a6bbdd646b214ea71b5bfbb0a55eb41eb5eae2/src/Common/src/CoreLib/System/Diagnostics/Tracing/TraceLogging/Statics.cs#L177
https://github.com/dotnet/corefx/blob/60a6bbdd646b214ea71b5bfbb0a55eb41eb5eae2/src/Common/src/CoreLib/System/Globalization/CalendarData.Unix.cs#L314
https://github.com/dotnet/corefx/blob/60a6bbdd646b214ea71b5bfbb0a55eb41eb5eae2/src/Common/src/CoreLib/System/Globalization/CultureData.cs#L2271
...

Reply:
Fixed, all except the 2nd one (`FixedBufferExtensions`) since it uses the length:
```csharp
int length = span.IndexOf('\0');
return length < 0 ? span.Length : length;
```